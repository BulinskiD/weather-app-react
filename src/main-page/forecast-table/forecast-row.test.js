import React from 'react';
import ReactDOM from 'react-dom';
import ForecastRow from './forecast-row';
import {shallow} from "enzyme";


describe('Forecast row', ()=>{

    const clickFn = jest.fn();
    const city = {id: 1, name: "Sth", temperature: "20.0" };
    const index = 1;
    const unit = 'imperial';

    it('should render correctly with given props', ()=>{
        const component = shallow(<ForecastRow index={index} city={city} unit={unit} onRemoveCity={clickFn} />);
        expect(component).toMatchSnapshot();
    });

    it('should call clickFn 1 times after Button click', ()=>{
        const component = shallow(<ForecastRow index={index} city={city} unit={unit} onRemoveCity={clickFn} />);
        component.find('Button').simulate('click');

        expect(clickFn).toBeCalledTimes(1);
    });
});
