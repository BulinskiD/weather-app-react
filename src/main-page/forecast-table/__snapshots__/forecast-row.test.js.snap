// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast row should render correctly with given props 1`] = `
<tr>
  <td
    className="text-center"
  >
    1
  </td>
  <td
    className="text-center"
  >
    <Link
      to="/details/1"
    >
      Sth
    </Link>
  </td>
  <td
    className="text-center"
  >
    20.0
     ° 
    F
  </td>
  <td
    className="text-center"
  >
    <Button
      active={false}
      className="col-12 bg-light text-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="danger"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f056",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
            ],
            "iconName": "minus-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        style={
          Object {
            "marginRight": "3%",
          }
        }
        symbol={false}
        title=""
        transform={null}
      />
      Usuń
    </Button>
  </td>
</tr>
`;
